provider/core/src/hello_world.rs

```diff
@@ -157,33 +156,49 @@ make_exportable_provider!(HelloWorldProvider, [HelloWorldV1Marker,]);
 ///     .take_payload()
 ///     .expect("Data should be present");
 ///
-/// assert_eq!(b"{\"message\":\"Hallo Welt\"}", german_hello_world.get());
-pub struct HelloWorldJsonProvider(HelloWorldProvider);
+/// assert_eq!(german_hello_world.get(), br#"{"message":"Hallo Welt"}"#);
+pub struct HelloWorldJsonProvider;
 
+#[cfg(feature = "deserialize_json")]
 impl BufferProvider for HelloWorldJsonProvider {
     fn load_buffer(
         &self,
         key: DataKey,
         req: DataRequest,
     ) -> Result<DataResponse<BufferMarker>, DataError> {
         key.match_key(HelloWorldV1Marker::KEY)?;
-        let result = self.0.load(req)?;
+        let result = HelloWorldProvider.load(req)?;
```

Comment:
Nit: Even though the old `self.0` stuff is technically not necessary, please keep this the way it was. There used to be a time that HelloWorldProvider contained an internally mutable map of strings.